### PR TITLE
Commited in the 47139_super_index

### PR DESCRIPTION
**Pull request by mdmahmudurrahman**

### CODE REVIEW

**Comment by manazirahsan** [comment]
Ok

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1,4 @@
+continue
+@users
+continue
+@users
```
Comment (shakil-fg) [inline .byebug_history]: Remove file, add to .gitignore

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -27,21 +27,27 @@
...
-                </div>
+                  <div class="form-group">
+                    <%= link_to t(:sign_out_text), destroy_user_session_path, class:"btn btn-default" %>
+                  </div>
               <% else %>
-                <div class="form-group">
-                <%= link_to t(:sign_out_text), destroy_user_session_path %>
-                </div>
+                  <div class="form-group">
+                    <%= link_to t(:sign_in_text), new_user_session_path, class:"btn btn-default"%>
```
Comment (shakil-fg) [inline app/views/layouts/_header.html.erb]: 2 spaces

File: app/views/supervisor/users/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<div class="container">
+  <h2 class="bg-primary">Supervisor index</h2>
```
Comment (shakil-fg) [inline app/views/supervisor/users/index.html.erb]: i18n

File: app/views/layouts/_header.html.erb (hunk 2 above)
Comment (noor-framgia) [inline app/views/layouts/_header.html.erb]: indent

File: app/views/supervisor/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                <td><%= u.email%></td>
+                <% if current_user == u %>
+                    <td><%= link_to t(:edit_supervisor_text), edit_user_registration_path(u), class: "btn btn-primary btn-xs"  %>
+                      <%= link_to t(:delete), supervisor_user_path(u), method: :delete,
+                                  class: "btn btn-danger btn-xs", data: { confirm: t(:confirm_message_delete) } %></td>
+                <% end %>
+              </tr>
+          <% end %>
+        </table>
+     </div>
```
Comment (noor-framgia) [inline app/views/supervisor/users/index.html.erb]: indent 2 space

File: app/controllers/supervisor/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+class Supervisor::UsersController < ApplicationController
+  before_action :load_user, only: [:show, :edit, :destroy]
+
+  def index
+    @users = User.where(role:"Supervisor")
```
Comment (chuanhtuan) [inline app/controllers/supervisor/users_controller.rb]: any where clause have to write in a scope

File: app/controllers/supervisor/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  def destroy
+   if @user.destroy
+     flash[:success] = t(:success_message_delete)
+     redirect_to supervisor_users_path
+   end
+  end
+
+  private
+    def load_user
+      @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/supervisor/users_controller.rb]: ()

File: app/views/supervisor/users/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<div class="container">
+  <h2 class="bg-primary"><%= t :supervisor_index_text %></h2>
+     <div>
+        <table class="table table-bordered">
+          <tr>
+            <th><%= t :name_text_supervisor %></th>
+            <th><%= t :email_text_supervisor %></th>
+            <th><%= t :action_text_supervisor %></th>
+          </tr>
+          <% @users.each do |u| %>
```
Comment (chuanhtuan) [inline app/views/supervisor/users/index.html.erb]: <% @users.each do |user| %>

File: app/models/user.rb [hunk 8]
```diff
@@ -21,6 +21,8 @@ class User < ActiveRecord::Base
 
   has_many :reports
 
+  scope :by_role, -> role_type { where(role_type) }
```
Comment (chuanhtuan) [inline app/models/user.rb]: please study about enum in active record

File: app/models/user.rb [hunk 9]
```diff
@@ -21,10 +21,10 @@ class User < ActiveRecord::Base
 
   has_many :reports
 
-  ROLE = {
-      trainee: "Trainee",
-      supervisor: "Supervisor"
-  }
-
-  enum role: { supervisor: "Supervisor", trainee: "Trainee" }
+  scope :by_role, -> role_type { where(role_type) }
```
Comment (chuanhtuan) [inline app/models/user.rb]: remove

**Pull request merged**